portfolio/src/main/java/com/google/sps/servlets/MessagesServlet.java

```diff
@@ -24,29 +38,56 @@
 import java.util.ArrayList;
 import java.util.List;
 
+import java.util.Date;
+
 @WebServlet("/messages")
 public class MessagesServlet extends HttpServlet {
     
-    private List<String> messages;
-
-    @Override
-    public void init(){
-        messages = new ArrayList<>();
-    }
+    private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException{
+        String languageCode = request.getQueryString().split("=")[1];
```

Comment:
Looking up this by position is pretty error prone. (Eg what if you add another parameter in?) Could you parse the querystring instead and look for "languageCode" in the result? (Something like https://stackoverflow.com/questions/13592236/parse-a-uri-string-into-name-value-collection)

Reply:
Yes, thank you for pointing that out. At second glance I did not realize that I can use my getParameter() function in doGet just like I use it in doPost.